model/riscv_types.sail

```diff
@@ -266,13 +266,14 @@ overload to_str = {exceptionType_to_str}
 /* trap modes */
 
 type tv_mode = bits(2)
-enum TrapVectorMode = {TV_Direct, TV_Vector, TV_Reserved}
+enum TrapVectorMode = {TV_Direct, TV_Vector, TV_Smclic, TV_Reserved}
 
 val trapVectorMode_of_bits : tv_mode -> TrapVectorMode
 function trapVectorMode_of_bits (m) =
   match (m) {
     0b00 => TV_Direct,
     0b01 => TV_Vector,
+    0b11 => TV_Smclic,
```

Comment:
Only if you have the CLIC extension. Plus this is for all xtvec, not just mtvec.

Reply:
I see this in the Priv spec: 

_The mtvec register must always be implemented, but can contain a read-only value. If mtvec is
writable, the set of values the register may hold can vary by implementation. The value in the
BASE field must always be aligned on a 4-byte boundary, and the MODE setting may impose
additional alignment constraints on the value in the BASE field._

So just as TV_Vector may not be supported by an implementation, TV_Smclic may not be supported by an implementation.  When @billmcspadden-riscv adds WARL config support to the riscv-sail model, TV_Smclic WARL support can be handled the same way as TV_Vector.  So I think it is incorrect to say "only if you have the CLIC extension".  CLIC extension is defined as just another mtvec mode which is labeled Reserved in the current priv spec.

Reply:
Well there's a bit of a distinction though. Direct and Vectored are both specified as what moved.BASE=0/1 mean, with >=2 reserved. Whereas this CLIC encoding is specified in its own extension on top of the base privileged spec. That is, the philosophical distinction is that vectored is part of the base privileged spec but may not be supported, whereas CLIC support is its own named extension on top.